engine/src/main/java/org/terasology/engine/network/internal/NetworkSystemImpl.java

```diff
@@ -826,7 +827,7 @@ private void processRemovedClient(Client client) {
         }
         clientList.remove(client);
         clientPlayerLookup.remove(client.getEntity());
-        logger.info("Client disconnected: {}", client.getName());
+        logger.atInfo().addArgument(() -> client.getName()).log("Client disconnected: {}");
```

Comment:
info-level and only called when client removed so impact neglectable.
```suggestion
        logger.info("Client disconnected: {}", client.getName()); //NOPMD
```